### PR TITLE
CAMS-746 remediate sync trustee appointments

### PR DESCRIPTION
**Pull request by mattstankey**
# Purpose

Bring the `sync-trustee-appointments` dataflow up to current CAMS standards so it follows the same architectural and observability patterns as other import dataflows.

# Major Changes

- Converts `SyncTrusteeAppointments` from an object-literal export to a class (`SyncTrusteeAppointmentsUseCase`) with constructor injection — factory calls moved from function bodies to the constructor
- Adds `SyncTrusteeAppointmentsStartMessage` type extending `StartMessage` with `reset`, `deleteAll`, `overrideRuntimeState`, and `flushQueues` flags; `handleStart` handles each flag in the correct order
- Adds `deleteAll()` method to the use case for the `deleteAll` flag path
- Replaces `buildStartQueueTimerTrigger` shared helper in `timerTrigger` with a direct handler that includes `startTrace`/`completeDataflowTrace` on all exit paths
- Fixes `afterEach` mock cleanup in the `downstream event emission` test suite — moves `vi.restoreAllMocks()` and `closeDeferred` into `beforeEach` to prevent stale mock leakage when a test throws

# Testing/Validation

Implemented with TDD. All existing use case tests updated for the new class API (54 tests). New smoke tests added for `handleStart` (8 tests) and `timerTrigger` (2 tests), covering the happy path, empty-event early return, each StartMessage flag, DLQ routing on failure, and missing runtime state. 70 tests total across both files, all passing.

# Notes

`resolveGroupMatchedProfessionalId` remains a standalone named export because `backfill-trustee-appointments-downstream.ts` imports it directly — it is not part of the class.

The `flushQueues` handler logs and exits cleanly but does not yet implement queue dumping to blob storage — that requires a separate implementation. The flag is scaffolded now so callers can pass it without the start message being silently ignored.

# Definition of Done:

- [ ] Code refactored for clarity: Developers can understand the work simply by reviewing the code
- [ ] Dependency rule followed: More important code doesn't directly depend on less important code
- [ ] Development debt eliminated: UX and code aligns to the team's latest understanding of the domain
- [ ] No regressions: Changes do not cause regression in related or unrelated areas of the application

## Summary by Sourcery

Refactor the sync-trustee-appointments dataflow into a class-based use case with improved start handling and observability, and extend the import function to support operational control flags and richer telemetry.

New Features:
- Introduce SyncTrusteeAppointmentsUseCase class encapsulating trustee appointment sync operations with constructor-injected dependencies.
- Add support for extended start messages including reset, deleteAll, overrideRuntimeState, and flushQueues flags to control sync behaviour.
- Expose a deleteAll operation for bulk removal of trustee appointment records before re-syncing.
- Implement a dedicated timerTrigger handler that enqueues START messages and records dataflow traces.

Enhancements:
- Move trustee appointment processing, runtime state management, and event retrieval logic from module-level functions into instance methods for better structure and testability.
- Update the page handler to use the new use case class while preserving existing matching and DLQ routing behaviour.
- Ensure runtime state overrides and reset behaviour are handled consistently when fetching appointment events.
- Improve error handling and logging for DXTR fetches, runtime state persistence, and delete-all operations.
- Adjust downstream event emission logic tests to avoid stale mocks by resetting and closing resources in beforeEach.

Tests:
- Update existing sync-trustee-appointments use case tests to target the new class API and constructor-injected dependencies.
- Add new tests for handleStart covering happy path, empty-event early exit, all operational flags, DLQ routing on failure, and missing runtime state cases.
- Add new tests for timerTrigger to verify successful start-message enqueueing and correct failure telemetry on errors.
- Refine downstream event emission tests to ensure proper mock cleanup across test cases.

## Summary by Sourcery

Refactor the sync-trustee-appointments dataflow into a class-based use case and extend the import pipeline to support richer start controls and telemetry-aligned triggering.

New Features:
- Introduce SyncTrusteeAppointmentsUseCase class encapsulating trustee appointment event retrieval, processing, runtime state persistence, and bulk deletion operations.
- Add SyncTrusteeAppointmentsStartMessage with lastSyncDate, reset, deleteAll, overrideRuntimeState, and flushQueues flags to control sync behaviour.
- Expose a deleteAll operation to remove all case appointment records before re-syncing.
- Implement a dedicated timerTrigger that enqueues START messages and records dataflow traces.

Enhancements:
- Move trustee appointment processing, runtime state management, and event retrieval from module-level functions into instance methods for improved structure and testability.
- Update handleStart and handlePage to use the new use case, honour operational flags, and ensure runtime state is only written when a latestSyncDate is available.
- Improve error handling and logging around DXTR fetches, runtime state persistence, delete-all failures, and trustee matching outcomes.
- Adjust trustee downstream event emission tests to reset mocks and close resources in beforeEach, avoiding stale mock leakage.

Tests:
- Update existing sync-trustee-appointments tests to target the new class API and constructor-injected dependencies.
- Add tests for handleStart covering flag combinations, empty-event early exit, DLQ routing on failures, and missing runtime state.
- Add tests for timerTrigger to verify successful start enqueueing and failure telemetry when context creation or output setting fails.
- Refine downstream event emission tests to assert audit logging and error handling across various matching scenarios.

### CODE REVIEW

File: backend/lib/use-cases/dataflows/sync-trustee-appointments.ts [hunk 1]
```diff
@@ -479,238 +438,319 @@ async function handleInactivePerfectMatch(
...
+        matchOutcome: 'error',
+        matchedTrusteeId: null,
+        scoringBreakdown: null,
+        appointmentStatus: null,
+      };
+
+      try {
+        const trusteeId = await matchTrusteeByName(context, event.dxtrTrustee.fullName);
+
+        const syncedCase = await this.casesRepo.getSyncedCase(event.caseId);
```
Comment (diabagatekelly) [inline backend/lib/use-cases/dataflows/sync-trustee-appointments.ts:548]: `processAppointments` fetches `syncedCase` here, but `applyResolvedTrustee` (line 169) fetches it again for the same `caseId`. That's two Cosmos reads per matched event — on the multiple-match path it's three (`resolveTrusteeWithFuzzyMatching` in the helpers file fetches it a third time too). Pass the already-fetched value in as a parameter:

```typescript
async function applyResolvedTrustee(
  context: ApplicationContext,
  event: TrusteeAppointmentSyncEvent,
  trusteeId: string,
  syncedCase: SyncedCase,  // caller passes this in
  casesRepo: CasesRepository,
  appointmentsRepo: TrusteeAppointmentsRepository,
): Promise<void> {
  // remove: const syncedCase = await casesRepo.getSyncedCase(event.caseId);
```

**Review by diabagatekelly — COMMENTED** [review]
**Overall:** Good refactor — the class conversion is clean, the `handleStart` flag dispatch is ordered correctly, and the use-case tests are thorough. A few things need attention before merge, and a few more are worth a follow-up issue.

**Before merge:**
- Double-fetch on `syncedCase` (inline comment at `sync-trustee-appointments.ts:545`) — real performance issue on larger batches.
- Null-safety gap in `applyResolvedTrustee` (line 170): the guard `if (syncedCase && ...)` acknowledges `syncedCase` can be falsy, but `syncedCase.courtDivisionCode`, `syncedCase.chapter`, etc. are dereferenced unconditionally at lines 196, 203, 215, 234, 261, 267, 279, 296–297 inside the feature-flag block. If the guard fires, those lines throw `TypeError` with no useful domain message. Either remove the guard and rely on `getSyncedCase` throwing `NotFoundError`, or add an explicit assertion before the downstream block: `if (!syncedCase) { throw new CamsError(MODULE_NAME, { message: \`syncedCase not found for ${event.caseId}\` }); }`
- `lastSyncDate` bracket notation (function-app line 106) — silent type bypass, inline comment there.
- Function-app test mocking pattern (test line 56) — prototype spy approach is fragile, inline comment there.

**Nice to have before merge:**
- `perfectMatchInactiveCount` is computed in `scenarioDistribution` but is not included in the `details` block emitted to App Insights in `handlePage` (lines 197–207 of the function-app file). The other six counters are there but this one is silently dropped.
- Feature flag string `'downstream-trustee-appointments-enabled'` appears twice (use-case lines 191 and 256) — worth a module-level constant since a typo produces `undefined` at runtime with no type error.
- `factory.getApiToDataflowsGateway(context)` is called twice inside `applyResolvedTrustee` (lines 218 and 281) — instantiate once. Same note applies to promoting `applyResolvedTrustee` and `handleInactivePerfectMatch` to private class methods: both accept repos as parameters that are already on `this`, and both call `factory.get*` inline, bypassing the constructor injection the class establishes.

**Good candidates for a follow-up beads issue:**
- `deleteAll` has no direct unit tests in the use-case test file — both the success and error-catch paths are untested at the unit level.
- Duplicate reVerification tests at lines 1273–1391 of the use-case test file — these repeat tests already in the `previously uncovered branches` block for IMPERFECT_MATCH, HIGH_CONFIDENCE_MATCH, and MULTIPLE_TRUSTEES_MATCH.
- `flushQueues` is scaffolded but unimplemented — worth a ticket so callers know what to expect when that feature lands.

File: backend/function-apps/dataflows/import/sync-trustee-appointments.ts [hunk 2]
```diff
@@ -65,9 +67,45 @@ async function handleStart(startMessage: StartMessage, invocationContext: Invoca
...
+          success: false,
+          error: deleteResult.error.message,
+        });
+        return;
+      }
+    }
+
+    const { events, latestSyncDate } = await useCase.getAppointmentEvents(
       context,
       startMessage['lastSyncDate'],
```
Comment (diabagatekelly) [inline backend/function-apps/dataflows/import/sync-trustee-appointments.ts]: `lastSyncDate` is not declared on `SyncTrusteeAppointmentsStartMessage` — bracket notation bypasses TypeScript's checker and silently evaluates to `undefined` if the field isn't present. (`sync-cases.ts:77` has the same pattern — both should be fixed.)

Add it to the type:
```typescript
type SyncTrusteeAppointmentsStartMessage = StartMessage & {
  lastSyncDate?: string;
  reset?: boolean;
  deleteAll?: boolean;
  overrideRuntimeState?: TrusteeAppointmentsSyncState;
  flushQueues?: boolean;
};
```
Then use `startMessage.lastSyncDate`.

File: backend/function-apps/dataflows/import/sync-trustee-appointments.test.ts [hunk 3]
```diff
@@ -51,20 +52,20 @@ describe('sync-trustee-appointments handlePage', () => {
       dlqMessages: [],
       scenarioDistribution: makeEmptyScenarioDistribution(),
     };
-    vi.spyOn(SyncTrusteeAppointmentsModule.default, 'processAppointments').mockResolvedValue(
-      processResult,
-    );
+    vi.spyOn(
+      SyncTrusteeAppointmentsModule.default.prototype,
```
Comment (diabagatekelly) [inline backend/function-apps/dataflows/import/sync-trustee-appointments.test.ts:56]: All the `handlePage` and `handleStart` tests mock the use case via `vi.spyOn(SyncTrusteeAppointmentsModule.default.prototype, ...)`. This prototype-patches the real production class — the tests never run any use-case logic. A contract change in `processAppointments` that doesn't update the mock goes undetected.

The use-case test file already provides comprehensive behavioral coverage at the right layer. For the function-app tests, the right boundary to mock is the factory — the same way the use-case tests do it:

```typescript
vi.spyOn(factory, 'getCasesRepository').mockReturnValue(mockCasesRepo as CasesRepository);
```

Or extract a typed interface and inject a stub. Either way, the function-app tests should focus on what's unique to this layer: queue wiring, telemetry shape, and retry/DLQ routing.

File: backend/function-apps/dataflows/import/sync-trustee-appointments.test.ts [hunk 4]
```diff
@@ -200,3 +206,297 @@ describe('sync-trustee-appointments handlePagePoison', () => {
...
+      courtId: '081',
+      dxtrTrustee: { fullName: 'John Doe' },
+    }) as TrusteeAppointmentSyncEvent;
+
+  beforeEach(() => {
+    vi.restoreAllMocks();
+  });
+
+  async function setupMocks(overrides?: {
+    getAppointmentEventsResult?: { events: TrusteeAppointmentSyncEvent[]; latestSyncDate: string };
```
Comment (diabagatekelly) [inline backend/function-apps/dataflows/import/sync-trustee-appointments.test.ts]: `latestSyncDate` is typed as `string` here, but this helper is called with `latestSyncDate: undefined` at lines 281, 337, and 439 — a type mismatch TypeScript should catch. The real return type of `getAppointmentEvents` has this field as `string | undefined`. The helper's type should match:

```typescript
getAppointmentEventsResult?: { events: TrusteeAppointmentSyncEvent[]; latestSyncDate: string | undefined };
```

File: backend/lib/use-cases/dataflows/sync-trustee-appointments.ts (hunk 1 above)
Comment (mattstankey) [reply backend/lib/use-cases/dataflows/sync-trustee-appointments.ts:548]: Fixed. Added `syncedCase` as a parameter to `applyResolvedTrustee` and removed the internal `casesRepo.getSyncedCase()` call. `processAppointments` now fetches it once and passes the value in.

File: backend/function-apps/dataflows/import/sync-trustee-appointments.ts (hunk 2 above)
Comment (mattstankey) [reply backend/function-apps/dataflows/import/sync-trustee-appointments.ts]: Fixed. Added `lastSyncDate?: string` to the type and switched to dot notation.

File: backend/function-apps/dataflows/import/sync-trustee-appointments.test.ts (hunk 4 above)
Comment (mattstankey) [reply backend/function-apps/dataflows/import/sync-trustee-appointments.test.ts]: Fixed. Changed to `string | undefined` to match the actual return type of `getAppointmentEvents`.

File: backend/function-apps/dataflows/import/sync-trustee-appointments.test.ts (hunk 3 above)
Comment (mattstankey) [reply backend/function-apps/dataflows/import/sync-trustee-appointments.test.ts:56]: Acknowledged, but keeping the current approach intentionally. The function-app tests exist to verify queue wiring, telemetry shape, and DLQ routing — the unique concerns of this layer. They are not intended to re-test use-case logic, which is already comprehensively covered by the use-case test file. Mocking at the prototype boundary (rather than the factory) keeps these tests focused and avoids re-implementing use-case test infrastructure at the wrong layer. A contract change to `processAppointments` that breaks the function-app behavior would surface in the use-case tests, which is the right place for that catch.

**Review by diabagatekelly — APPROVED** [review]

**Review by btposey — APPROVED** [review]
